examples/solvers/adupdates_tomography.py

```diff
@@ -0,0 +1,149 @@
+                  for i in range(SPLIT_NUMBER)]
+else:
+    raise ValueError('unknown data split`{}`: typo?'.format(SPLIT_METHOD))
+
+# Create the artificial data.
+data_spaces = [op.range for op in ray_trafos]
+data = [op(phantom) + 5.0 * odl.phantom.white_noise(ds)
+        for (op, ds) in zip(ray_trafos, data_spaces)]
+
+# Functionals and operators for the total variation. This is the l1 norm of the
+# (discretized) gradient of the reconstruction. For each of the dimensions
+# we create two functionals and two operators.
+
+# Start with empty lists ...
+tv_functionals = []
+tv_operators = []
+tv_stepsizes = []
+
+# ... and for each dimension of the reconstruction space ...
+reco_shape = reconstruction_space.shape
+reco_dim = len(reco_shape)
+for j in range(reco_dim):
+    # ... add two operators taking only the even and odd elements,
+    # respectively, in that dimension.
+    partial_der = odl.PartialDerivative(
+        reconstruction_space, j, pad_mode='order0')
+    allpoints = list(np.ndindex(reco_shape))
+    sampling_points1 = [list(p) for p in allpoints if p[j] % 2 == 0]
+    sampling_points1 = np.array(sampling_points1).T.tolist()
+    sampling_points2 = [list(p) for p in allpoints if p[j] % 2 == 1]
+    sampling_points2 = np.array(sampling_points2).T.tolist()
+    op1 = reconstruction_space.cell_sides[j] * odl.SamplingOperator(
+        reconstruction_space, sampling_points1) * partial_der
+    op2 = reconstruction_space.cell_sides[j] * odl.SamplingOperator(
+        reconstruction_space, sampling_points2) * partial_der
+    tv_functionals += [odl.solvers.L1Norm(op1.range),
+                       odl.solvers.L1Norm(op2.range)]
+    tv_operators += [op1, op2]
+    tv_stepsizes += [0.5 / reco_shape[j], 0.5 / reco_shape[j]]
+
+# Functional and operator enforcing the nonnegativity of the image.
+nonneg_functional = odl.solvers.IndicatorNonnegativity(reconstruction_space)
+nonneg_operator = odl.IdentityOperator(reconstruction_space)
+nonneg_stepsize = 1.0
+
+# ... and the data fit functionals. The coefficient is a regularization
+# paratemeter, which determines the tradeoff between data fit and regularity.
+data_fit_functionals = [1.0 *
+                        odl.solvers.L2NormSquared(ds).translated(rhs)
+                        for (ds, rhs) in zip(data_spaces, data)]
+# In the stepsizes, we avoid the possible division by zero by adding a small
+# positive value. The matrix corresponding to the operator `op` has only
+# nonnegative entries, which makes this ensures that the final results are
+# positve.
+data_fit_stepsizes = [1.0 / (1e-6 + op(op.adjoint(ds.one())))
+                      for (ds, op) in zip(data_spaces, ray_trafos)]
+# Alternative choice without vector-valued stepsizes could be
+# data_fit_stepsizes = [1.0 / op.norm(estimate=True) ** 2 for op in ray_trafos]
+
+# Now we build up the ingredients of our algorithm:
+# Start at a zero image, ...
+x = reconstruction_space.zero()
+
+# collect all the functionals, ...
```

Comment:
Above you also used `...` at the front, which made the continuation more obvious.